### PR TITLE
Top n queries configuration page

### PR DESCRIPTION
**Pull request by LilyCaroline17**
### Description
Configuration page of top n queries dashboard in the query insights dashboard plugin. User can edit N (number of queries collected within the window size) and the window size (time duration where the top n queries are collected). For N, values have to be between 1 and 100. For window size, time has to be in (1, 5, 10, 30) minutes or (1-24) hours.
Screenshot 2024-07-12 at 4 38 06 PM

### Issues Resolved
Related to https://github.com/opensearch-project/query-insights-dashboards/issues/5

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,4 +8,5 @@ yarn-error.log
 .DS_Store
 /cypress/screenshots/
 /cypress/videos/
-target
\ No newline at end of file
+target
+.eslintcache
```
Comment (ansjcy) [inline .gitignore:12]: You might need to remove this once the overview page is merged, otherwise there will be conflicts. The same applies for opensearch_dashboards.json and package.json.

File: public/pages/Configuration/Configuration.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      max={24}
+      required={true}
+      value={windowSize}
+      onChange={(e) => onWindowSizeChange(e)}
+    />
+  );
+
+  const WindowChoice = time === timeUnits[0].value ? MinutesBox : HoursBox;
+
+  let changed;
```
Comment (ansjcy) [inline public/pages/Configuration/Configuration.tsx:78]: Is this a string or a boolean? seems like you are using it as a boolean?
If so we can use const instead
```
const isChanged = topNSize !== defaultTopN || windowSize !== defaultWindowSize
```

File: public/pages/Configuration/Configuration.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    setTime(e.target.value);
+  };
+
+  const MinutesBox = () => (
+    <EuiSelect
+      id="minutes"
+      required={true}
+      options={minutesOptions}
+      value={windowSize}
+      onChange={(e) => onWindowSizeChange(e)}
```
Comment (ansjcy) [inline public/pages/Configuration/Configuration.tsx:62]: we can simplify this with
```
onChange={onWindowSizeChange}
```

File: public/pages/Configuration/Configuration.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    />
+  );
+
+  const HoursBox = () => (
+    <EuiFieldNumber
+      min={1}
+      max={24}
+      required={true}
+      value={windowSize}
+      onChange={(e) => onWindowSizeChange(e)}
```
Comment (ansjcy) [inline public/pages/Configuration/Configuration.tsx:72]: same as above

File: public/pages/Configuration/Configuration.tsx [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+  const WindowChoice = time === timeUnits[0].value ? MinutesBox : HoursBox;
+
+  let changed;
+  if (topNSize !== defaultTopN) {
+    changed = 'topN';
+  } else if (windowSize !== defaultWindowSize) {
+    changed = 'windowSize';
+  }
+
+  let valid = false;
```
Comment (ansjcy) [inline public/pages/Configuration/Configuration.tsx:85]: we can declare valid as const to avoid accidentally modifying it later.
```
const isValid = (() => {
    const nVal = parseInt(topNSize, 10);
    if (nVal < 1 || nVal > 100) return false;
    if (time === timeUnits[0].value) return true;
    const windowVal = parseInt(windowSize, 10);
    return windowVal >= 1 && windowVal <= 24;
  })();
```

File: public/pages/TopNQueries/TopNQueries.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
+import React from 'react';
```
Comment (ansjcy) [inline public/pages/TopNQueries/TopNQueries.tsx:1]: This file shouldn't be included in this PR, you might need to do a rebase after the overview page got merged.

File: public/pages/Configuration/Configuration.tsx (hunk 2 above)
Comment (LilyCaroline17) [reply public/pages/Configuration/Configuration.tsx]: It depends on the metric, and also can change each render which is why it's not a constant.

File: public/pages/TopNQueries/TopNQueries.tsx (hunk 6 above)
Comment (LilyCaroline17) [reply public/pages/TopNQueries/TopNQueries.tsx]: This is a file that does redirecting to the three pages, so it should exist and then the changes for each page will merge.

**Review by ansjcy — APPROVED** [review]
